### PR TITLE
React to `HtmlEncodedString` rename to `HtmlString`.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Removed the Mvc instance of `HtmlString` since it now exists in `HtmlAbstractions`.

#4558
aspnet/HtmlAbstractions#25

/cc @rynowak @dougbu @pranavkm

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs [hunk 1]
```diff
@@ -14,18 +14,18 @@ namespace Microsoft.AspNetCore.Mvc.TagHelpers.Cache
...
+                throw new ArgumentNullException(nameof(context));
             }
-            
-            if (context.Html == null) 
+
+            if (context.Html == null)
             {
-                throw new ArgumentNullException(nameof(context.Html));    
+                throw new ArgumentNullException(nameof(context.Html));
             }
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs]: This isn't the right exception/message

File: test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ImageTagHelperTest.cs [hunk 2]
```diff
@@ -142,7 +143,7 @@ public void PreservesOrderOfSourceAttributesWhenRun()
             Assert.Equal(expectedOutput.TagName, output.TagName);
             Assert.Equal(4, output.Attributes.Count);
 
-            for(int i=0; i < expectedOutput.Attributes.Count; i++)
+            for (int i = 0; i < expectedOutput.Attributes.Count; i++)
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/ImageTagHelperTest.cs]: var

**Comment by rynowak** [comment]
Is this namespace included with a default `using` in razor? should it be now?

**Comment by rynowak** [comment]
:shipit:

**Comment by NTaylorMullen** [comment]
> Is this namespace included with a default using in razor? should it be now?

It's not; however, I don't think it should be. If a user is creating `HtmlString`s in a view they're doing some really odd things :smile:.

**Comment by rynowak** [comment]
eh, I'd be really surprised if we don't get feedback on compat about this one. Obiv `Html.Raw` is there, but I'm gonna bet `HtmlString` shows up in lots of peoples codez.

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs]: /cc @sebastienros I remember having the same comment when you wrote the code but there was a reason. What was the reason?

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs [hunk 3]
```diff
@@ -14,18 +14,13 @@ namespace Microsoft.AspNetCore.Mvc.TagHelpers.Cache
     /// </summary>
     public class DistributedCacheTagHelperFormatter : IDistributedCacheTagHelperFormatter
     {
-        
+
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs]: Remove?

**Comment by pranavkm** [comment]
Would be nice to follow up on @rynowak 's suggestion on adding it to the default view namespaces.

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs [hunk 4]
```diff
@@ -14,18 +14,13 @@ namespace Microsoft.AspNetCore.Mvc.TagHelpers.Cache
...
         /// <inheritdoc />
         public Task<byte[]> SerializeAsync(DistributedCacheTagHelperFormattingContext context)
         {
-            if (context == null) 
-            {
-                throw new ArgumentNullException(nameof(context));    
-            }
-            
-            if (context.Html == null)
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/DistributedCacheTagHelperFormatter.cs]: `context.Html == null` is still an error case. Will now `throw` a useless NRE.

Instead fix the `Exception` to be an `ArgumentException` and use our usual `FormatPropertyOfTypeCannotBeNull` resource.

**Comment by dougbu** [comment]
I agree w/ @pranavkm and @rynowak's point about adding the new `HtmlString` to the default namespaces we use automatically in views.

I'm more concerned about user's tag helpers, HTML helper extensions, and similar e.g. this change broke a number of the MVC tag helpers. Not sure scaffolding needs to include `using Microsoft.AspNetCore.Html;` in every controller class. But this definitely needs an Announcement.

**Comment by dougbu** [comment]
:shipit:

**Comment by Eilon** [comment]
I seriously doubt almost anyone uses the `HtmlString` type directly from a CSHTML page. There's almost no good reason to use that type there.

From HTML helpers, sure, but there are much bigger changes one has to make from a CS file anyway. And with VS it's just a <kbd>Ctrl</kbd>+<kbd>.</kbd> away to get the namespace.

**Comment by javiercn** [comment]
Agree with @NTaylorMullen we shouldn't encourage people to new up IHtmlString on their views

**Pull request merged**